src/dotnet/commands/dotnet-migrate/MigrateCommand.cs

```diff
@@ -76,12 +86,57 @@ public int Execute()
 
             WriteReport(migrationReport);
 
+            MoveProjectJsonArtifactsToBackup(_doBackup, migrationReport);
+
             return migrationReport.FailedProjectsCount;
         }
 
-        private void WriteReport(MigrationReport migrationReport)
+        private void MoveProjectJsonArtifactsToBackup(bool doBackup, MigrationReport migrationReport)
         {
+            if (!doBackup)
+            {
+                return;
+            }
+            
+            if (migrationReport.FailedProjectsCount > 0)
+            {
+                return;
+            }
+            
+            _backupDirectory.Create();
+
+            var globalJson = Path.Combine(_workspaceDirectory.FullName, GlobalSettings.FileName);
+
+            if (File.Exists(globalJson))
+            {
+                File.Move(globalJson, Path.Combine(_backupDirectory.FullName, GlobalSettings.FileName));
+            }
+            
+            foreach (var report in migrationReport.ProjectMigrationReports)
+            {
+                var projectDirectory = PathUtility.EnsureTrailingSlash(report.ProjectDirectory);
+                
+                var relativeDirectory = PathUtility.GetRelativePath(PathUtility.EnsureTrailingSlash(_workspaceDirectory.FullName), projectDirectory);
+
+                var targetDirectory = String.IsNullOrEmpty(relativeDirectory)
+                    ? _backupDirectory.FullName
+                    :  Path.Combine(_backupDirectory.FullName, relativeDirectory);
+
+                PathUtility.EnsureDirectory(PathUtility.EnsureTrailingSlash(targetDirectory));
 
+                var movableFiles = new DirectoryInfo(projectDirectory)
+                    .EnumerateFiles()
+                    .Where(f => f.Name == Project.FileName || f.Extension == ".xproj");
+                
+                foreach (var movableFile in movableFiles)
+                {
+                    movableFile.MoveTo(Path.Combine(targetDirectory, movableFile.Name));
+                }
+            }
+        }
+
+        private void WriteReport(MigrationReport migrationReport)
```

Comment:
Do we want to tell the user that a backup was created and where it can be found?